featurebyte/service/observation_table.py

```diff
@@ -481,7 +486,19 @@ async def get_observation_table_task_payload(
             document=FeatureByteBaseDocumentModel(_id=output_document_id, name=data.name),
         )
 
-        if isinstance(data.request_input, BaseRequestInput):
+        if isinstance(data.request_input, ObservationTableObservationInput):
+            source_observation_table = await self.get_document(
+                document_id=data.request_input.observation_table_id
+            )
+            # primary entity, context, use case ids and target namespace id are
+            # inherited from source observation table and will be populated in the task
+            data.primary_entity_ids = source_observation_table.primary_entity_ids
+            data.context_id = None
+            data.use_case_id = None
```

Comment:
Why do we inherit only `primary_entitiy_ids` but not `context_id` and `use_case_id`? It is because the task only populates the latter two but not `primary_entity_ids`?

Reply:
I think either way is fine. The current comment seems a bit misleading to mention that primary entity will be populated in the task. Would be good to document this behaviour clearly in the comment.